src/main/java/net/sf/jabref/logic/cleanup/ISSNCleanup.java

```diff
@@ -0,0 +1,33 @@
+package net.sf.jabref.logic.cleanup;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.model.FieldChange;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class ISSNCleanup implements CleanupJob {
+
+    private static final Pattern ISSN_PATTERN_NODASH = Pattern.compile("^(\\d{4})(\\d{3}[\\dxX])$");
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        Optional<String> issn = entry.getFieldOptional("issn");
+        if (!issn.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        Matcher matcher = ISSN_PATTERN_NODASH.matcher(issn.get());
```

Comment:
I would move this functionality to a separate `ISSN` class which does the parsing. For example, the code then might look as follows:
````
ISSN issn = new ISSN(issnText);
if (issn.isValid()) {
   entry.setField("issn", issn.ToString())
}
````
Then you could reuse the same logic to add a ISSN integrity check. The same remark applies to your other ISBN PR.